aws/lambda-with-other-services/lambda-dynamodb/lambda-dynamodb.md

```diff
@@ -1,3 +1,298 @@
+
+```
+$ pnpm install uuid
+```
+
+Create an index.js file in the project and open it in your editor. This is where we will be writing the function itself.
+
+First add the dependencies. We need to import the uuid package we've just installed. We also need to require the AWS SDK (which all Lambda functions have access to, so we don't have to install it). The SDK will then allow us to create an instance of the DynamoDB document client.
+
+```javascript
+const AWS = require('aws-sdk');
+const docClient = new AWS.DynamoDB.DocumentClient({region: 'us-west-2'});
+const uuid4 = require('uuid/v4');
+```
+
+This docClient instance is a standard way to target DynamoDB. It provides a simple way to create JavaScript sets. As we are specifying a region here, our function will be able to write to the table even if they are in different AWS regions.
+
+Now for the function itself. As with all Lambda functions, we will use an ```exports.handler``` function that has three arguments: event, context, and callback.
+
+```javascript
+exports.handler = function(event, context, callback) {
+
+	}
+}
+```
+
+First we need to specify the parameters we will passing into the DynamoDB docClient method. This includes an ```Item``` object for the table entry itself. All entries must have an id as that is the primary key, and because it is a movies table each entry will have a name and a score. The id is created by calling the uuid function we have added, and the other two values will come from the event object. Besides ```Item,``` we also need a ```TableName``` key, the value of which will naturally be the name of our table.
+
+```javascript
+exports.handler = function(event, context, callback) {
+	const uuid = uuid4();
+	const params = {
+		Item: {
+			id: uuid,
+			name: event.name,
+			score: event.score
+		},
+		TableName: 'movies'
+	}
+}
+```
+
+Next we'll pass those params into the docClient method. In docClient we'll then call a function to handle the result. This is where the callback parameter of our main handler function comes into play. Callback() is a method available to Lambda functions that has two arguments: the first is an error, and the second is non-error data. Therefore, if docClient.put() returns an error, we call callback() with the error as the first parameter and the second as null. If there is no error, we have null as the first argument and the return as the second one.
+
+```javascript
+docClient.put(params, function(error, data){
+	if (error) {
+		callback(error, null);
+	} else {
+		callback(null, data);
+	}
+});
+```
+
+Here is the entirety of index.js as we have it so far:
+
+```javascript
```

Comment:
Make sure you update this based on the changes I asked for above. Include the promisified version of `docClient.put()`